File: 0111-minimum-depth-of-binary-tree/0111-minimum-depth-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int minDepth(TreeNode root) {
        return minHeight(root);
    }
    public int minHeight(TreeNode root)
    {
        if(root == null) return 0;
        
        if(root.left == null) return 1 + minHeight(root.right);
        if(root.right == null) return 1 + minHeight(root.left);
        
        return 1 + Math.min(minHeight(root.left), minHeight(root.right));
        
    }
}

File: 0111-minimum-depth-of-binary-tree/0111-minimum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def minDepth(self, root: Optional[TreeNode]) -> int:
        if not root:return 0
        
        mini = float('inf')
        def helper(node, depth):
            nonlocal mini
            if not node.left and not node.right:
                mini = min(mini, depth)
                
            if node.left:
                helper(node.left, depth+1)
                
            if node.right:
                helper(node.right, depth+1)
                
            return 
        helper(root, 1)
        return mini
        
        